docs/checks.md

```diff
@@ -2,159 +2,479 @@
-If the project is hosted on GitHub, You can learn more about dependencies for projects using [GitHub dependency graph](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph). 
+Risk: `Medium` (possible compromised dependencies)
+
+This check tries to determine if the project is an application that has declared
+and pinned its dependencies. A "pinned dependency" is a dependency that is
+explicitly set to a specific version instead of allowing a range of versions. It
+is currently limited to repositories hosted on GitHub, and does not support
+other source hosting repositories (i.e., Forges). If this project is a library
+(not an application), this check should automatically pass (but see limitations
+below).
+
+The check works by looking for:
+
+  - the following files in the root directory: go.mod, go.sum (Golang),
+    package-lock.json, npm-shrinkwrap.json (Javascript), requirements.txt,
+    pipfile.lock (Python), gemfile.lock (Ruby), cargo.lock (Rust), yarn.lock
+    (package manager), composer.lock (PHP), vendor/, third_party/, third-party/; 
+  - unpinned dependencies in Dockerfiles, shell scripts and GitHub workflows. 
+
+Pinned dependencies reduce several security risks:
+
+  - They ensure that checking and deployment are all done with the same
+    software, reducing deployment risks, simplifying debugging, and enabling
+    reproducibility. 
+  - They can help mitigate compromised dependencies from undermining the
+    security of the project (in the case where you've evaluated the pinned
+    dependency, you are confident it's not compromised, and a later version is
+    released that is compromised). 
+  - They are one way to
+    [counter dependency confusion (aka substitution) attacks](https://azure.microsoft.com/en-us/resources/3-ways-to-mitigate-risk-using-private-package-feeds/),
+    in which an application uses multiple feeds to acquire software packages (a
+    "hybrid configuration"), and attackers fool the user into using a malicious
+    package via a feed that was not expected for that package.
+
+However, pinning dependencies can inhibit software updates, either because of a
+security vulnerability or because the pinned version is compromised. Mitigate
+this risk by:
+
+  - [having applications and _not_ libraries pin to specific versions](https://jbeckwith.com/2019/12/18/package-lock/);
+  - using automated tools to notify applications when their dependencies are
+    outdated;
+  - using applications that _do_ pin dependencies update quickly.
```

Comment:
Just wanted to highlight this line to be double-checked for accuracy and clarity.

Reply:
Thanks. Line 394 is updated.